src/v2/view-builder/views/device/DeviceChallengePollView.js

```diff
@@ -17,6 +18,12 @@ const request = (opts) => {
   return $.ajax(ajaxOptions);
 };
 
+const cancelPolling = (appState, triggeredByUser = false) => {
```

Comment:
make `cancelPolling` part of `polling.js`.. it has access to `this.option.Appstate`... 

`actionParams` can be part as args.. 
```
cancelPolling(actionParams) {
this.stopPolling(); // check if this is needed?? to stopPoll before trigger cancel 

this.options.appState.trigger('invokeAction', 'cancel', actionParams);
}
```

Reply:
@pradeepdewda-okta , `stopPolling` is to stop/remove the ongoing `polling` request on the page, `cancelPolling` is a different API which responds with a different remediation. When this new remediation is returned, the current UI will be removed and `stopPolling` will be invoked. I know the naming is confusing, `cancelPolling` is simply named after its API URL. We can name it something else, such as `stopCurrentAuthentication`. 
In regarding of moving `cancelPolling` to polling.js, I can think about that and potentially address it in a separate JIRA. We have another [JIRA](https://oktainc.atlassian.net/browse/OKTA-390965) for refactoring that's targeted for next week.

Reply:
>When this new remediation is returned, the current UI will be removed and stopPolling will be invoked.

Nice that make sense.

It seems like a polling action.. so we can rename as `invokePollingActionWithParams`  and move it to `polling.js` (in your next PR)

Reply:
> > When this new remediation is returned, the current UI will be removed and stopPolling will be invoked.
> 
> Nice that make sense.
> 
> It seems like a polling action.. so we can rename as `invokePollingActionWithParams` and move it to `polling.js` (in your next PR)

Let me give more thought about the naming. Thanks.